skimage/filters/thresholding.py

```diff
@@ -766,27 +766,36 @@ def _mean_std(image, w):
            Retrieval XV, (San Jose, USA), Jan. 2008.
            DOI:10.1117/12.767755
     """
-    if w == 1 or w % 2 == 0:
+    if isinstance(w, int):
+        w1 = w
+        w2 = w
+        kern = np.zeros((w2 + 1,) * image.ndim)
+
+    if isinstance(w, tuple) and len(w) == image.ndim:
+        w1 = w[0] * w[1]
+        w2 = max(w)
+        kern = np.zeros(w)
+
+    if w1 == 1 or w1 % 2 == 0:
```

Comment:
Oh! Clever way to check it with the product! But then don't use `w[0] * w[1]`, which only works for 2D images. Instead, use `np.prod(w)`. And call it something different than `w1`, which is not very descriptive. Maybe `total_window_size`?